src/Emulator/Extensions/Mocks/HPSHostController.cs

```diff
@@ -171,6 +174,114 @@ private byte[] GetBytesFromSlave(int count, TimeInterval timeInterval)
             return result;
         }
 
+        private void PollForError(TimeInterval timeInterval){
+            var errors = ReadCommonErrorStatusBytes(timeInterval);
+            var sw = new Stopwatch();
+            sw.Start();
+            while (errors[0] == 0x00 && sw.Elapsed < new TimeSpan(0,0,3)){
+                errors = ReadCommonErrorStatusBytes(timeInterval);
+            }
+        }
+
+        // When we need to read for more than 255 bytes, we will need this method
+        private byte[] GetLongBytes(int count, TimeInterval timeInterval) {            
+            var result = new byte[count];
+            int rounds = count /255;
+            int remainder = count % 255; 
+            for (int i = 0; i <= rounds; i ++) {
+                int toRead = i == rounds ? remainder : 255;
+                Array.Copy(GetBytesFromSlave(toRead, timeInterval), 0, result, i * 255, toRead);
+            }
+            return result;
+        }
+
+        public void StartSocketServer(TimeInterval timeInterval)
+        {  
+            Console.WriteLine(currentSlave.GetType().Name);
+            string path = "/tmp/i2c.sock";
+            if (File.Exists(path))
+            {
+                File.Delete(path);
+            }
+            var socket = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified);
+            socket.Bind(new UnixEndPoint(path));
+            socket.Listen(0);
+            Console.WriteLine("Server started waiting for client to connect...");
+            Console.WriteLine("Starting on: "+ path);
+            currentSlave.Read(0);
+            var connection = socket.Accept();
+            Console.WriteLine("Connection established");
+
+            var commandBuffer = new byte[1];
+            int length;
+            while(SocketConnected(connection)){
+                try {
+                    PollForSocketData(connection, (int)numOfBytes.MessageType, commandBuffer);
+                }
+                catch(Exception e)
+                {
+                    Console.WriteLine("{0} Exception caught.", e);
+                    continue;
+                }
+                switch(commandBuffer[0])
+                {
+                    case (byte)SocketI2CMessageType.Stop:
+                        continue;
```

Comment:
Does this mean that we're ignoring the stop condition? Our firmware does use the stop condition in some cases - from memory, it's used when we write to flash - so for those cases to work, we'd the Stop condition to be plumbed through to the firmware

Reply:
our way of addressing this would be to call `currentSlave.FinishTransmission`. It's not a perfect approach and our modelling on the I2C controller side might need slight improvements as well, but I'll go with that

Reply:
It seems that in the STM32F7_I2C.cs, this method is not really implemented? I'll call it anyway?